### PR TITLE
metrics notes

### PR DESCRIPTION
**Pull request by shwstppr**
Notes about VM and VM disk metrics and the improvements added in https://github.com/apache/cloudstack/pull/6803

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan docbuild

**Comment by rohityadavcloud** [comment]
@blueorangutan docbuild

**Review by rohityadavcloud — APPROVED** [review]

**Comment by DaanHoogland** [comment]
@shwstppr the images do not show in the paragraph they are intended for (or am I looking with crooked eyes/false expectations?)
see [the paragraphs](https://qa.cloudstack.cloud/builds/docs-build/pr/294/adminguide/virtual_machines.html?highlight=virtual%20machines%20metric#vm-disk-metrics)

**Comment by shwstppr** [comment]
Thanks @DaanHoogland for pointing it out. Made the changes
@blueorangutan docbuild

**Comment by andrijapanicsb** [comment]
@shwstppr we fixed graphs in the meantime - so these need to be updated here please (new graphs on the volume level)

**Comment by shwstppr** [comment]
@blueorangutan docbuild

**Comment by shwstppr** [comment]
@andrijapanicsb @borisstoyanov are you good on this?

File: source/adminguide/virtual_machines.rst [hunk 1]
```diff
@@ -1281,6 +1281,42 @@ CloudStack follows the below sequence of operations to provide GPU/vGPU support
...
+setting vm.stats.interval with a default of 60000 milliseconds).
+VM statistics include include compute, storage and network statistics.
+
+VM statistics are stored in the database for historical data for a desired time period. These historical statistics then can be retrieved using listVirtualMachinesUsageHistory API. For system VMs, the same historical stats can be retrieved using listSystemVmsUsageHistory API
+
+VM statistics retention time in the database is controlled by the global configuration - `vm.stats.max.retention.time`. Default value is 720 minutes, i.e., 12 hours. Other global configurations that affects virtual machine statistics is:
+
+- `vm.stats.user.vm.only` - When set to 'false' stats for system VMs will be collected otherwise stats collection will be done only for user VMs.
+
+In UI, historical VM statistics are shown in the Metrics tab in an individual VM view, as shown in the image below.
```
Comment (DaanHoogland) [inline source/adminguide/virtual_machines.rst:1297]:
```suggestion
In the UI, historical VM statistics are shown in the Metrics tab in an individual VM view, as shown in the image below.
```

File: source/adminguide/virtual_machines.rst [hunk 2]
```diff
@@ -1281,6 +1281,42 @@ CloudStack follows the below sequence of operations to provide GPU/vGPU support
...
+
+|vm-metrics-ui.png|
+
+
+VM Disk Metrics
+---------------
+
+Similar to VM statistics, VM disk statistics can also be collected on a regular interval (defined by global setting vm.disk.stats.interval with a default value of 0 seconds which disables VM disk stats collection). VM disk stats are collected in form of diskiopstotal, diskioread, diskiowrite, diskkbsread and diskkbswrite.
```
Comment (DaanHoogland) [inline source/adminguide/virtual_machines.rst:1305]:
```suggestion
Similar to VM statistics, VM disk statistics (disk stats) can also be collected on a regular interval (defined by global setting vm.disk.stats.interval with a default value of 0 seconds which disables disk stats collection). Disk stats are collected in form of diskiopstotal, diskioread, diskiowrite, diskkbsread and diskkbswrite.
```

File: source/adminguide/virtual_machines.rst [hunk 3]
```diff
@@ -1281,6 +1281,42 @@ CloudStack follows the below sequence of operations to provide GPU/vGPU support
...
+
+VM disk statistics can also be made to store in the database and the historical statistics can be retrieved using listVolumesUsageHistory API.
+
+VM disk statistics retention in the database is controlled by the global configuration - `vm.disk.stats.retention.enabled`. Default value is false, i.e., retention of VM disk statistics is disabled. Other global configurations that affects virtual machine disk statistics are:
+
+- `vm.disk.stats.interval.min` - Minimal interval (in seconds) to report vm disk statistics. If vm.disk.stats.interval is smaller than this, use this to report vm disk statistics.
+
+- `vm.disk.stats.max.retention.time` - The maximum time (in minutes) for keeping VM disk stats records in the database. The VM disk stats cleanup process will be disabled if this is set to 0 or less than 0.
```
Comment (DaanHoogland) [inline source/adminguide/virtual_machines.rst:1313]:
```suggestion
- `vm.disk.stats.max.retention.time` - The maximum time (in minutes) for keeping disk stats records in the database. The disk stats cleanup process will be disabled if this is set to 0 or less than 0.
```

**Review by DaanHoogland — APPROVED** [review]
content looks good, some style suggestion

**Comment by shwstppr** [comment]
@blueorangutan docbuild

**Review by andrijapanicsb — APPROVED** [review]
LGTM

**Pull request merged**